pkg/controller/containerjfr/containerjfr_controller_test.go

```diff
@@ -0,0 +1,563 @@
+				Expect(deployment.ObjectMeta.Labels).To(Equal(expectedDeployment.ObjectMeta.Labels))
+				Expect(deployment.ObjectMeta.Annotations).To(Equal(expectedDeployment.ObjectMeta.Annotations))
+				Expect(deployment.Spec.Selector).To(Equal(expectedDeployment.Spec.Selector))
+
+				// compare Pod template
+				template := deployment.Spec.Template
+				expectedTemplate := expectedDeployment.Spec.Template
+				Expect(template.ObjectMeta).To(Equal(expectedTemplate.ObjectMeta))
+				Expect(template.Spec.Containers).To(Equal(expectedTemplate.Spec.Containers))
+				Expect(template.Spec.Volumes).To(Equal(expectedTemplate.Spec.Volumes))
+			})
+		})
+		Context("after containerjfr reconciled successfully", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewContainerJFR(),
+				}
+			})
+			It("should be idempotent", func() {
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+				ReconcileFully(client, *controller, false)
+
+				obj := &rhjmcv1beta1.ContainerJFR{}
+				err := client.Get(context.Background(), req.NamespacedName, obj)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Reconcile again
+				result, err := controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(result).To(Equal(reconcile.Result{}))
+
+				obj2 := &rhjmcv1beta1.ContainerJFR{}
+				err = client.Get(context.Background(), req.NamespacedName, obj2)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(obj2.Status).To(Equal(obj.Status))
+				Expect(obj2.Spec).To(Equal(obj.Spec))
+			})
+		})
+		Context("After a minimal containerjfr reconciled successfully", func() {
+			BeforeEach(func() {
+				objs = []runtime.Object{
+					test.NewMinimalContainerJFR(),
+				}
+			})
+			It("should be idempotent", func() {
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "containerjfr", Namespace: "default"}}
+				ReconcileFully(client, *controller, true)
+
+				obj := &rhjmcv1beta1.ContainerJFR{}
+				err := client.Get(context.Background(), req.NamespacedName, obj)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Reconcile again
+				result, err := controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(result).To(Equal(reconcile.Result{}))
+
+				obj2 := &rhjmcv1beta1.ContainerJFR{}
+				err = client.Get(context.Background(), req.NamespacedName, obj2)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(obj2.Status).To(Equal(obj.Status))
+				Expect(obj2.Spec).To(Equal(obj.Spec))
+			})
+		})
+		Context("ContainerJFR does not exist", func() {
+			It("Should do nothing", func() {
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: "does-not-exist", Namespace: "default"}}
+				result, err := controller.Reconcile(req)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(result).To(Equal(reconcile.Result{}))
+			})
+		})
+
+	})
+})
+
+func NewFakeSecret(name string) *corev1.Secret {
```

Comment:
Please make the helper functions begin with a lower-case letter to make them private visibility.